src/Server.php

```diff
@@ -1500,6 +1500,16 @@ public function shutdown() : void{
 		if($this->isRunning){
 			$this->isRunning = false;
 			$this->signalHandler->unregister();
+
+			if(TimingsHandler::isEnabled()){
+				TimingsHandler::createReportFile($this->getDataPath() . DIRECTORY_SEPARATOR . "timings")->onCompletion(
+					function(string $timingsFile) : void{
+						$this->logger->info($this->language->translate(KnownTranslationFactory::pocketmine_command_timings_timingsWrite($timingsFile)));
+						TimingsHandler::setEnabled(false);
+					},
+					fn() => throw new \AssertionError("This promise is not expected to be rejected")
```

Comment:
This doesn't seem like a good idea. The promise could be rejected if the file already exists.